backend/utils.py

```diff
@@ -164,3 +164,10 @@ def wrapper(*args, **kwargs):
 
 def get_id_from_token(token):
     return User.auth.get_user(token).user.id
+
+def extract_keywords(resume_text, keywords):
+    """
+    Extracts keywords that appear in the resume text.
+    Case-insensitive.
+    """
+    return [kw for kw in keywords if kw.lower() in resume_text.lower()]
```

Comment:
This function is simple and concise, making it easy to understand and use. You could optimize for large inputs by initially converting resume_text to lowercase once before starting the search:
```suggestion
def extract_keywords(resume_text, keywords):
    """
    Extracts keywords that appear in the resume text.
    Case-insensitive.
    """
    lower_resume_text = resume_text.lower()
    return [kw for kw in keywords if kw.lower() in lower_resume_text]
```

Reply:
This has the same problem --- you should probably start by removing "stop words", and ranking words by number of occurrences.